### PR TITLE
EMSUSD-1135 Add flag to export empty transform

### PR DESCRIPTION
**Pull request by pierrebai-adsk**
- Add a USD export job argument called includeEmptyTransforms.
- It defaults to true.
- Add a flag with the same name to the MayaUSD export command, with short flag named "-iet".
- Document the new flag in the command readme.
- Make it readable in the JobExportArgs Python class.
- Implement post-export pruning of empty Xform and Scope.
- Keep default prim even if empty
- To be empty it has to have no children, no payload and no reference.
- Add two unit tests, one including and one excluding empty transforms.
- Add unit tests to verify that prims with payloads or refs are not considered empty.
- Add unit test to verify that empty non Xform and non Scope are kept.
- Add a unit test for the case the default prim is empty.

### CODE REVIEW

**Review by AramAzhari-adsk — APPROVED** [review]
Looks good to me.

File: lib/mayaUsd/fileio/jobs/writeJob.cpp [hunk 1]
```diff
@@ -780,6 +783,81 @@ TfToken UsdMaya_WriteJob::_WriteVariants(const UsdPrim& usdRootPrim)
...
+            defaultPrimPath = SdfPath(std::string("/") + mJobCtx.mArgs.defaultPrim);
+        else
+            defaultPrimPath = SdfPath(mJobCtx.mArgs.defaultPrim);
+    }
+
+    std::vector<SdfPath> toRemove;
+
+    for (UsdPrim prim : mJobCtx.mStage->Traverse()) {
+        if (defaultPrimPath != prim.GetPath())
+            if (isEmptyPrim(prim))
```
[the remark this answers is not in the document]
Comment (pierrebai-adsk) [reply lib/mayaUsd/fileio/jobs/writeJob.cpp:846]: Well, if the prim is empty, it has no children so there is nothing to optimize anyway :)

File: test/lib/usd/plugin/plugInfoExportEmptyConfig.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "Plugins": [
+    {
+      "Info": {
+        "UsdMaya": {
+            "UsdExport": {
+                "chaser": ["AddPayloadOrRefChaser"]
```
[the remark this answers is not in the document]
Comment (pierrebai-adsk) [reply test/lib/usd/plugin/plugInfoExportEmptyConfig.json:7]: Sorry, I don't understand the question. What does cover "Maya Material to USD Prim for Chasers" means?

File: lib/mayaUsd/fileio/jobs/writeJob.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (AramAzhari-adsk) [reply lib/mayaUsd/fileio/jobs/writeJob.cpp:846]: Fair enough!

File: test/lib/usd/plugin/plugInfoExportEmptyConfig.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (AramAzhari-adsk) [reply test/lib/usd/plugin/plugInfoExportEmptyConfig.json:7]: I responded to this privately.
